### PR TITLE
Setup tests. Add basic GenericRestClient tests

### PR DESCRIPTION
**Pull request by a-tarasyuk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,20 +4,41 @@
...
+    "test:watch": "npm run clean && karma start"
   },
   "dependencies": {
-    "@types/lodash": "^4.14.62",
-    "assert": "^1.4.1",
-    "lodash": "4.x",
-    "synctasks": "^0.3.0"
+    "@types/lodash": "4.14.107",
+    "assert": "1.4.1",
+    "lodash": "4.17.5",
```
Comment (deregtd) [inline package.json]: Need to leave this as 4.x, or at least ^4.17.0 -- you're prescribing a specific lodash here.

File: package.json [hunk 2]
```diff
@@ -4,20 +4,41 @@
...
+    "synctasks": "0.3.1"
```
Comment (deregtd) [inline package.json]: same here -- the ^ is intentional so it can pick up updates

File: package.json [hunk 3]
```diff
@@ -4,20 +4,41 @@
...
+    "test": "npm run clean && karma start --singleRun",
+    "test:watch": "npm run clean && karma start"
   },
   "dependencies": {
-    "@types/lodash": "^4.14.62",
-    "assert": "^1.4.1",
-    "lodash": "4.x",
-    "synctasks": "^0.3.0"
+    "@types/lodash": "4.14.107",
+    "assert": "1.4.1",
```
Comment (deregtd) [inline package.json]: ditto

File: package.json (hunk 1 above)
Comment (a-tarasyuk) [reply package.json]: thanks, will change

File: src/ExponentialTime.ts [hunk 4]
```diff
@@ -83,3 +86,29 @@ export class ExponentialTime {
         return res;
     }
 }
+
+
```
Comment (deregtd) [inline src/ExponentialTime.ts]: need to remove all this test code

**Pull request merged**